Switch to GitHub Actions

jparise opened the pull request.

jparise committed:
Switch to GitHub Actions

elprans commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,31 @@
+        pip install mock
+    - name: Tests
+      run: |
+        python setup.py test
```
`sudo` might still be required here

jparise replied:
That didn't seem to help. 😞

elprans commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,31 @@
+      if: matrix.python-version == '2.7'
+      run: |
+        python -m pip install --upgrade pip
+        pip install mock
```
Remove "mock" from `tests_require` in setup.py, I think that's what trips some of the tests up.

jparise commented:
I understand the test failures now. They're failing because the current Python + OS environment results in `openat` syscalls when Python's `open()` is used.

I'll work on updating the tests.

jparise committed:
Recognize 'openat' as a valid open() syscall

In our current test environment, the Python open() function results in
the 'openat' syscall (rather than 'open').

The test suite has been updated to only recognize 'openat' in the
specific cases. It will take a little more work to support both 'open'
and 'openat' because the position of the `pathname` argument differs.

elprans reviewed (approved):
> They're failing because the current Python + OS environment results in `openat` syscalls when Python's `open()` is used.

Yep, good find! Glibc switched to using the `openat` syscall for `open()` by default [in 2.26](https://sourceware.org/git/?p=glibc.git;a=commit;h=b41152d716ee9c5ba34495a54e64ea2b732139b5)
